integration_tests/test_systems_real_api.py

```diff
@@ -0,0 +1,270 @@
+import json
+import time
+
+import docker
+import jcapiv1
+import os
+
+from click.testing import CliRunner
+from jcapiv1 import SystemsApi
+from integration_tests import API_KEY
+from jccli import cli
+
+
+JC_CONNECT_KEY = os.getenv('JC_CONNECT_KEY')
+DOCKER_IMAGE = 'sagejcdevs/jumpcloud-agent:v0.0.2'  # Name of image to spin up
+SYSTEM_COUNT = 5  # The number of docker containers to spin up and connect to the cloud
+
+
+class TestSystemsRealApi:
+    @classmethod
+    def setup_class(cls):
+        '''
+        Spin up docker images and check that they have connected
+        '''
+
+        apiv1_configuration = jcapiv1.Configuration()
+        apiv1_configuration.api_key['x-api-key'] = API_KEY
+        cls.systems_api = SystemsApi(jcapiv1.ApiClient(configuration=apiv1_configuration))
+
+        current_systems = cls.systems_api.systems_list(
+            content_type='application_json',
+            accept='application/json'
+        )
+        assert current_systems.total_count == 0
+
+        docker_client = docker.from_env()
+        cls.containers = []
+        for i in range(SYSTEM_COUNT):
+            container = docker_client.containers.run(
+                DOCKER_IMAGE,
+                environment={'JC_CONNECT_KEY': JC_CONNECT_KEY},
+                detach=True,
+                stdin_open=True,
+                tty=True
+            )
+            cls.containers.append(container)
+
+        # Wait for containers to boot up and connect to JumpCloud
+        while True:
+            current_systems = cls.systems_api.systems_list(
+                content_type='application/json',
+                accept='application/json'
+            )
+            print("number of reporting systems on JumpCloud:")
+            print(current_systems.total_count)
+            if current_systems.total_count == SYSTEM_COUNT:
+                break
+            time.sleep(1)
+
+        # Check whether system hostnames have loaded (this takes a while after they phone home)
+        while True:
```

Comment:
This and the above loop have no exits if the the external containers aren't successfully launched, and since it looks like in Travis the tests are not capturing output and there is a print in the loops they could continue to run for a long time if not without Travis shutting them down for lack of output. Wouldn't really hurt to put these in a `for ... else` loop with a reasonable number of cycles and fail if the else is reached w/o break.

Reply:
Added 300 cycle (= 5 minutes + execution time) limit for each one. Does that length sound good?

Reply:
Yeah seems fine. It's in a test so doesn't need to be super complex but probably worth having to avoid consuming a Travis runner for a long time if there is some issue with starting up the containers.